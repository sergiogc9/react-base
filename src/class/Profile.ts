export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
