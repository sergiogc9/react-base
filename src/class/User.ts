export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
