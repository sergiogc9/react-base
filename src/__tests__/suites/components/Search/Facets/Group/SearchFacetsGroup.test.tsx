import { shallow } from "enzyme";
import React from "react";
import { RouterState } from "react-router-redux";

import SearchFacetsGroup from "@src/components/Search/Facets/Group/SearchFacetsGroup";
import GA from '@src/lib/googleAnalytics';
import { FacetItem } from "@src/class/Facet";
import { FiltersFacetGroup } from "@src/class/Filter";
import { Facet } from '@src/class/Facet';

const facetItems: FacetItem[] = [{ key: "24", counter: 22 }, { key: "30", counter: 50 }];
const facetItemsTags: FacetItem[] = [{ key: "tag1", counter: 10 }, { key: "tag2", counter: 20 }];
const mediaItemsTags: FacetItem[] = [{ key: "22222", counter: 100, name: "media 2", type: "news" }, { key: "11111", counter: 20, name: "media 1", type: "print" }];
const languageItemsTags: FacetItem[] = [{ key: "127", counter: 1 }];

const facetsFiltersGroups: FiltersFacetGroup = {
	"channel_type_id": [{ key: "24" }],
	"tenants.tags": [{ key: "tag1" }],
	"media_id": [{ key: "333333", name: "Media 3", type: "news" }]
};

const routerLocation: RouterState["location"] = {
	pathname: "/article",
	search: "",
	state: "",
	hash: ""
};

let onToggleFacetFilter: any;
let onRemoveFacetFilterGroup: any;
let onToggleFacetGroupOpened: any;
let onSetShowMoreDialogFacetGroupKey: any;

describe("<SearchFacetsGroup />", () => {
	let wrapper: any;

	beforeAll(() => {
		window.matchMedia = jest.fn((query: string) => ({
			matches: query.indexOf('(min-width: 1025px)') !== -1
		}));
	});

	beforeEach(() => {
		onToggleFacetFilter = jest.fn();
		onRemoveFacetFilterGroup = jest.fn();
		onToggleFacetGroupOpened = jest.fn();
		onSetShowMoreDialogFacetGroupKey = jest.fn();
		wrapper = shallow(<SearchFacetsGroup
			location={routerLocation}
			feedType={null}
			facetFilters={facetsFiltersGroups}
			facetsGroupsOpened={["channel_type_id", "tenants.tags"]}
			onToggleFacetFilter={onToggleFacetFilter}
			onRemoveFacetFilterGroup={onRemoveFacetFilterGroup}
			onToggleFacetGroupOpened={onToggleFacetGroupOpened}
			onSetShowMoreDialogFacetGroupKey={onSetShowMoreDialogFacetGroupKey}
			groupKey="channel_type_id"
			groupItems={facetItems}
			loadingFacets={false}
		/>);
	});

	it("Test component snapshot with group opened", () => {
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot with group not opened", () => {
		wrapper.setProps({ facetsGroupsOpened: [] });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot with not filter enabled", () => {
		wrapper.setProps({ facetFilters: {} });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot with group not enabled", () => {
		wrapper.setProps({ groupItems: [], facetFilters: {} });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot with tenants.tags group", () => {
		wrapper.setProps({ groupKey: "tenants.tags", groupItems: facetItemsTags });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot with media_id group", () => {
		wrapper.setProps({ groupKey: "media_id", groupItems: mediaItemsTags });
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it("Test component snapshot with language.id group", () => {
		wrapper.setProps({ groupKey: "language.id", groupItems: languageItemsTags });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot with a facet filtered not received by query", () => {
		wrapper.setProps({ groupItems: [] });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot in preview page with group without tooltip and feed online", () => {
		wrapper.setProps({ feedType: "online", location: { ...routerLocation, pathname: "/focus/focusId/feed/create/socialmedia" } });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot in preview page with group without tooltip", () => {
		wrapper.setProps({ feedType: "socialmedia", location: { ...routerLocation, pathname: "/focus/focusId/feed/create/socialmedia" } });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot in preview page with language group key", () => {
		wrapper.setProps({ groupKey: "language_id", feedType: "socialmedia", location: { ...routerLocation, pathname: "/focus/focusId/feed/create/socialmedia" } });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test component snapshot in preview page with country group key", () => {
		wrapper.setProps({ groupKey: "country_path", feedType: "socialmedia", location: { ...routerLocation, pathname: "/focus/focusId/feed/create/socialmedia" } });
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test group item checkbox change event", () => {
		GA.trackEvent = jest.fn();
		expect(onToggleFacetFilter).toHaveBeenCalledTimes(0);
		expect(GA.trackEvent).toHaveBeenCalledTimes(0);
		wrapper.find('#searchFacetsGroupItemCheckbox-channel_type_id-24').at(0).props().onChange();
		expect(onToggleFacetFilter).toHaveBeenCalledTimes(1);
		expect(GA.trackEvent).toHaveBeenCalledTimes(1);
		wrapper.find('#searchFacetsGroupItemName-channel_type_id-24').at(0).simulate('click');
		expect(onToggleFacetFilter).toHaveBeenCalledTimes(2);
		expect(GA.trackEvent).toHaveBeenCalledTimes(2);
	});

	it("Test toggle group opened handler with group enabled", () => {
		expect(onToggleFacetGroupOpened).toHaveBeenCalledTimes(0);
		wrapper.find('#searchFacetsGroup-channel_type_id .search-facets-group-header').at(0).simulate('click');
		expect(onToggleFacetGroupOpened).toHaveBeenCalledTimes(1);
	});

	it("Test toggle group opened handler with group disabled", () => {
		wrapper.setProps({ groupItems: [], facetFilters: {} });
		expect(onToggleFacetGroupOpened).toHaveBeenCalledTimes(0);
		wrapper.find('#searchFacetsGroup-channel_type_id .search-facets-group-header').at(0).simulate('click');
		expect(onToggleFacetGroupOpened).toHaveBeenCalledTimes(0);
	});

	it("Test remove group enabled filters handler", () => {
		expect(onRemoveFacetFilterGroup).toHaveBeenCalledTimes(0);
		wrapper.find('#searchFacetsGroup-channel_type_id .search-facets-group-remove-filters').at(0).simulate('click');
		expect(onRemoveFacetFilterGroup).toHaveBeenCalledTimes(1);
	});

	it("Test on set show more dialog click handler", () => {
		GA.trackEvent = jest.fn();
		expect(onSetShowMoreDialogFacetGroupKey).toHaveBeenCalledTimes(0);
		expect(GA.trackEvent).toHaveBeenCalledTimes(0);
		wrapper.find('#searchFacetsGroupSetShowMore-channel_type_id').at(0).simulate('click');
		expect(onSetShowMoreDialogFacetGroupKey).toHaveBeenCalledTimes(1);
		expect(GA.trackEvent).toHaveBeenCalledTimes(1);
	});

	it("Test untagged document tag special render", () => {
		const newTagItems = [{ key: Facet.untaggedDocumentKey, counter: 10 }];
		wrapper.setProps({ groupKey: "tenants.tags", groupItems: newTagItems });
		expect(wrapper.find('.search-facets-group-item-name .custom-facet').exists()).toBe(true);
	});

	it("Test facet counters should not be displayed while loading facets", () => {
		expect(wrapper.find(".counter").exists()).toEqual(true);
		wrapper.setProps({ loadingFacets: true });
		expect(wrapper.find(".counter").exists()).toEqual(false);
	});
});
