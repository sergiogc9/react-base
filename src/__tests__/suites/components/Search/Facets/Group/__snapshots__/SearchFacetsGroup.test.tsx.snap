// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchFacetsGroup /> Test component snapshot in preview page with country group key 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsGroup-country_path\\"><div class=\\"search-facets-group-header \\"><i class=\\"md-icon material-icons\\">keyboard_arrow_right</i><span>filters.country.value</span><button type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-btn--tooltip md-inline-block search-facets-group-help-tooltip\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-tooltip-container\\"><span aria-hidden=\\"true\\"></span></div><i class=\\"md-icon material-icons md-text--inherit\\">help_outline</i></button></div></div>"`;

exports[`<SearchFacetsGroup /> Test component snapshot in preview page with group without tooltip 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsGroup-channel_type_id\\"><div class=\\"search-facets-group-header enabled\\"><i class=\\"md-icon material-icons\\">keyboard_arrow_down</i><span>filters.channel.value</span><div class=\\"search-facets-group-remove-filters\\" title=\\"filters.reset.value\\"><i class=\\"md-icon material-icons\\">close</i></div></div><div class=\\"search-facets-group-items\\"><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container checkbox-green\\"><input type=\\"checkbox\\" id=\\"searchFacetsGroupItemCheckbox-channel_type_id-24\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsGroupItemCheckboxs-channel_type_id[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsGroupItemCheckbox-channel_type_id-24\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><div id=\\"searchFacetsGroupItemName-channel_type_id-24\\" class=\\"search-facets-group-item-name\\"><span>filters.channel.24</span><span> - <span class=\\"counter\\">22</span></span></div></div><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container\\"><input type=\\"checkbox\\" id=\\"searchFacetsGroupItemCheckbox-channel_type_id-30\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsGroupItemCheckboxs-channel_type_id[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsGroupItemCheckbox-channel_type_id-30\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><div id=\\"searchFacetsGroupItemName-channel_type_id-30\\" class=\\"search-facets-group-item-name\\"><span>filters.channel.30</span><span> - <span class=\\"counter\\">50</span></span></div></div><div id=\\"searchFacetsGroupSetShowMore-channel_type_id\\" class=\\"search-facets-group-item\\"><i class=\\"md-icon material-icons\\">add</i>filters.show_more</div></div></div>"`;

exports[`<SearchFacetsGroup /> Test component snapshot in preview page with group without tooltip and feed online 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsGroup-channel_type_id\\"><div class=\\"search-facets-group-header enabled\\"><i class=\\"md-icon material-icons\\">keyboard_arrow_down</i><span>filters.channel.value</span><div class=\\"search-facets-group-remove-filters\\" title=\\"filters.reset.value\\"><i class=\\"md-icon material-icons\\">close</i></div></div><div class=\\"search-facets-group-items\\"><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container checkbox-green\\"><input type=\\"checkbox\\" id=\\"searchFacetsGroupItemCheckbox-channel_type_id-24\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsGroupItemCheckboxs-channel_type_id[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsGroupItemCheckbox-channel_type_id-24\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><div id=\\"searchFacetsGroupItemName-channel_type_id-24\\" class=\\"search-facets-group-item-name\\"><span>filters.channel.24</span><span> - <span class=\\"counter\\">22</span></span></div></div><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container\\"><input type=\\"checkbox\\" id=\\"searchFacetsGroupItemCheckbox-channel_type_id-30\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsGroupItemCheckboxs-channel_type_id[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsGroupItemCheckbox-channel_type_id-30\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><div id=\\"searchFacetsGroupItemName-channel_type_id-30\\" class=\\"search-facets-group-item-name\\"><span>filters.channel.30</span><span> - <span class=\\"counter\\">50</span></span></div></div><div id=\\"searchFacetsGroupSetShowMore-channel_type_id\\" class=\\"search-facets-group-item\\"><i class=\\"md-icon material-icons\\">add</i>filters.show_more</div></div></div>"`;

exports[`<SearchFacetsGroup /> Test component snapshot in preview page with language group key 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsGroup-language_id\\"><div class=\\"search-facets-group-header \\"><i class=\\"md-icon material-icons\\">keyboard_arrow_right</i><span>filters.language.value</span><button type=\\"button\\" class=\\"md-btn md-btn--icon md-pointer--hover md-btn--tooltip md-inline-block search-facets-group-help-tooltip\\"><div class=\\"md-ink-container\\"></div><div class=\\"md-tooltip-container\\"><span aria-hidden=\\"true\\"></span></div><i class=\\"md-icon material-icons md-text--inherit\\">help_outline</i></button></div></div>"`;

exports[`<SearchFacetsGroup /> Test component snapshot with a facet filtered not received by query 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsGroup-channel_type_id\\"><div class=\\"search-facets-group-header enabled\\"><i class=\\"md-icon material-icons\\">keyboard_arrow_down</i><span>filters.channel.value</span><div class=\\"search-facets-group-remove-filters\\" title=\\"filters.reset.value\\"><i class=\\"md-icon material-icons\\">close</i></div></div><div class=\\"search-facets-group-items\\"><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container checkbox-green\\"><input type=\\"checkbox\\" id=\\"searchFacetsGroupItemCheckbox-channel_type_id-24\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsGroupItemCheckboxs-channel_type_id[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsGroupItemCheckbox-channel_type_id-24\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><div id=\\"searchFacetsGroupItemName-channel_type_id-24\\" class=\\"search-facets-group-item-name\\"><span>filters.channel.24</span></div></div><div id=\\"searchFacetsGroupSetShowMore-channel_type_id\\" class=\\"search-facets-group-item\\"><i class=\\"md-icon material-icons\\">add</i>filters.show_more</div></div></div>"`;

exports[`<SearchFacetsGroup /> Test component snapshot with group not enabled 1`] = `"<div class=\\"search-facets-group disabled\\" id=\\"searchFacetsGroup-channel_type_id\\"><div class=\\"search-facets-group-header \\"><i class=\\"md-icon material-icons\\">keyboard_arrow_right</i><span>filters.channel.value</span></div></div>"`;

exports[`<SearchFacetsGroup /> Test component snapshot with group not opened 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsGroup-channel_type_id\\"><div class=\\"search-facets-group-header enabled\\"><i class=\\"md-icon material-icons\\">keyboard_arrow_right</i><span>filters.channel.value</span><div class=\\"search-facets-group-remove-filters\\" title=\\"filters.reset.value\\"><i class=\\"md-icon material-icons\\">close</i></div></div></div>"`;

exports[`<SearchFacetsGroup /> Test component snapshot with group opened 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsGroup-channel_type_id\\"><div class=\\"search-facets-group-header enabled\\"><i class=\\"md-icon material-icons\\">keyboard_arrow_down</i><span>filters.channel.value</span><div class=\\"search-facets-group-remove-filters\\" title=\\"filters.reset.value\\"><i class=\\"md-icon material-icons\\">close</i></div></div><div class=\\"search-facets-group-items\\"><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container checkbox-green\\"><input type=\\"checkbox\\" id=\\"searchFacetsGroupItemCheckbox-channel_type_id-24\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsGroupItemCheckboxs-channel_type_id[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsGroupItemCheckbox-channel_type_id-24\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><div id=\\"searchFacetsGroupItemName-channel_type_id-24\\" class=\\"search-facets-group-item-name\\"><span>filters.channel.24</span><span> - <span class=\\"counter\\">22</span></span></div></div><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container\\"><input type=\\"checkbox\\" id=\\"searchFacetsGroupItemCheckbox-channel_type_id-30\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsGroupItemCheckboxs-channel_type_id[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsGroupItemCheckbox-channel_type_id-30\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><div id=\\"searchFacetsGroupItemName-channel_type_id-30\\" class=\\"search-facets-group-item-name\\"><span>filters.channel.30</span><span> - <span class=\\"counter\\">50</span></span></div></div><div id=\\"searchFacetsGroupSetShowMore-channel_type_id\\" class=\\"search-facets-group-item\\"><i class=\\"md-icon material-icons\\">add</i>filters.show_more</div></div></div>"`;

exports[`<SearchFacetsGroup /> Test component snapshot with language.id group 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsGroup-language.id\\"><div class=\\"search-facets-group-header \\"><i class=\\"md-icon material-icons\\">keyboard_arrow_right</i><span>filters.language.value</span></div></div>"`;

exports[`<SearchFacetsGroup /> Test component snapshot with media_id group 1`] = `
"<div className=\\"search-facets-group \\" id=\\"searchFacetsGroup-media_id\\">
  <div className=\\"search-facets-group-header enabled\\" onClick={[Function: onClick]}>
    <FontIcon iconClassName=\\"material-icons\\">
      keyboard_arrow_right
    </FontIcon>
    <span>
      filters.media.value
    </span>
    <div className=\\"search-facets-group-remove-filters\\" onClick={[Function: onClick]} title=\\"filters.reset.value\\">
      <FontIcon iconClassName=\\"material-icons\\">
        close
      </FontIcon>
    </div>
  </div>
  <Collapse collapsed={true} animate={true} springConfig={{...}} minHeight={0}>
    <div className=\\"search-facets-group-items\\">
      <div className=\\"search-facets-group-item\\">
        <SelectionControl id=\\"searchFacetsGroupItemCheckbox-media_id-333333\\" name=\\"searchFacetsGroupItemCheckboxs-media_id[]\\" label=\\"\\" type=\\"checkbox\\" checked={true} className=\\"checkbox-green\\" onChange={[Function: onChange]} checkedCheckboxIcon={{...}} uncheckedCheckboxIcon={{...}} checkedRadioIcon={{...}} uncheckedRadioIcon={{...}} />
        <div id=\\"searchFacetsGroupItemName-media_id-333333\\" className=\\"search-facets-group-item-name\\" onClick={[Function: onClick]}>
          <FontIcon className=\\"search-facets-group-item-icon\\" iconClassName=\\"icon-news\\" />
          <span className={[undefined]}>
            Media 3
          </span>
        </div>
      </div>
      <div className=\\"search-facets-group-item\\">
        <SelectionControl id=\\"searchFacetsGroupItemCheckbox-media_id-22222\\" name=\\"searchFacetsGroupItemCheckboxs-media_id[]\\" label=\\"\\" type=\\"checkbox\\" checked={false} className=\\"\\" onChange={[Function: onChange]} checkedCheckboxIcon={{...}} uncheckedCheckboxIcon={{...}} checkedRadioIcon={{...}} uncheckedRadioIcon={{...}} />
        <div id=\\"searchFacetsGroupItemName-media_id-22222\\" className=\\"search-facets-group-item-name\\" onClick={[Function: onClick]}>
          <FontIcon className=\\"search-facets-group-item-icon\\" iconClassName=\\"icon-news\\" />
          <span className={[undefined]}>
            media 2
          </span>
          <span>
             - 
            <span className=\\"counter\\">
              100
            </span>
          </span>
        </div>
      </div>
      <div className=\\"search-facets-group-item\\">
        <SelectionControl id=\\"searchFacetsGroupItemCheckbox-media_id-11111\\" name=\\"searchFacetsGroupItemCheckboxs-media_id[]\\" label=\\"\\" type=\\"checkbox\\" checked={false} className=\\"\\" onChange={[Function: onChange]} checkedCheckboxIcon={{...}} uncheckedCheckboxIcon={{...}} checkedRadioIcon={{...}} uncheckedRadioIcon={{...}} />
        <div id=\\"searchFacetsGroupItemName-media_id-11111\\" className=\\"search-facets-group-item-name\\" onClick={[Function: onClick]}>
          <FontIcon className=\\"search-facets-group-item-icon\\" iconClassName=\\"icon-print\\" />
          <span className={[undefined]}>
            media 1
          </span>
          <span>
             - 
            <span className=\\"counter\\">
              20
            </span>
          </span>
        </div>
      </div>
      <div id=\\"searchFacetsGroupSetShowMore-media_id\\" className=\\"search-facets-group-item\\" onClick={[Function: onClick]}>
        <FontIcon iconClassName=\\"material-icons\\">
          add
        </FontIcon>
        filters.show_more
      </div>
    </div>
  </Collapse>
</div>"
`;

exports[`<SearchFacetsGroup /> Test component snapshot with not filter enabled 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsGroup-channel_type_id\\"><div class=\\"search-facets-group-header \\"><i class=\\"md-icon material-icons\\">keyboard_arrow_down</i><span>filters.channel.value</span></div><div class=\\"search-facets-group-items\\"><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container\\"><input type=\\"checkbox\\" id=\\"searchFacetsGroupItemCheckbox-channel_type_id-24\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsGroupItemCheckboxs-channel_type_id[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsGroupItemCheckbox-channel_type_id-24\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><div id=\\"searchFacetsGroupItemName-channel_type_id-24\\" class=\\"search-facets-group-item-name\\"><span>filters.channel.24</span><span> - <span class=\\"counter\\">22</span></span></div></div><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container\\"><input type=\\"checkbox\\" id=\\"searchFacetsGroupItemCheckbox-channel_type_id-30\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsGroupItemCheckboxs-channel_type_id[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsGroupItemCheckbox-channel_type_id-30\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><div id=\\"searchFacetsGroupItemName-channel_type_id-30\\" class=\\"search-facets-group-item-name\\"><span>filters.channel.30</span><span> - <span class=\\"counter\\">50</span></span></div></div><div id=\\"searchFacetsGroupSetShowMore-channel_type_id\\" class=\\"search-facets-group-item\\"><i class=\\"md-icon material-icons\\">add</i>filters.show_more</div></div></div>"`;

exports[`<SearchFacetsGroup /> Test component snapshot with tenants.tags group 1`] = `"<div class=\\"search-facets-group \\" id=\\"searchFacetsGroup-tenants.tags\\"><div class=\\"search-facets-group-header enabled\\"><i class=\\"md-icon material-icons\\">keyboard_arrow_down</i><span>filters.tags</span><div class=\\"search-facets-group-remove-filters\\" title=\\"filters.reset.value\\"><i class=\\"md-icon material-icons\\">close</i></div></div><div class=\\"search-facets-group-items\\"><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container checkbox-green\\"><input type=\\"checkbox\\" id=\\"searchFacetsGroupItemCheckbox-tenants.tags-tag1\\" checked=\\"\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsGroupItemCheckboxs-tenants.tags[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsGroupItemCheckbox-tenants.tags-tag1\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"true\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--theme-secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box</i></div></label></div><div id=\\"searchFacetsGroupItemName-tenants.tags-tag1\\" class=\\"search-facets-group-item-name\\"><span>tag1</span><span> - <span class=\\"counter\\">10</span></span></div></div><div class=\\"search-facets-group-item\\"><div class=\\"md-selection-control-container\\"><input type=\\"checkbox\\" id=\\"searchFacetsGroupItemCheckbox-tenants.tags-tag2\\" class=\\"md-selection-control-input\\" name=\\"searchFacetsGroupItemCheckboxs-tenants.tags[]\\" aria-hidden=\\"true\\"/><label for=\\"searchFacetsGroupItemCheckbox-tenants.tags-tag2\\" class=\\"md-selection-control-label md-pointer--hover md-text\\"><div role=\\"checkbox\\" aria-checked=\\"false\\" aria-labelledby=\\"\\" class=\\"md-fake-btn md-pointer--hover md-fake-btn--no-outline md-selection-control-toggle md-btn md-btn--icon md-text--secondary\\" tabindex=\\"0\\" aria-pressed=\\"false\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons md-text--inherit\\">check_box_outline_blank</i></div></label></div><div id=\\"searchFacetsGroupItemName-tenants.tags-tag2\\" class=\\"search-facets-group-item-name\\"><span>tag2</span><span> - <span class=\\"counter\\">20</span></span></div></div><div id=\\"searchFacetsGroupSetShowMore-tenants.tags\\" class=\\"search-facets-group-item\\"><i class=\\"md-icon material-icons\\">add</i>filters.show_more</div></div></div>"`;
